content/blog/writing-crossguard-policies/index.md

```diff
@@ -0,0 +1,167 @@
+    description: "Detailed monitoring should be enabled for all EC2 instances.",
+    enforcementLevel: "mandatory",
+    validateResource: validateTypedResource(aws.ec2.Instance, (instance, args, reportViolation) => {
+        if (instance.monitoring !== true) {
+            reportViolation("EC2 Instance should have monitoring enabled.");
+        }
+    }),
+};
+```
+
+If you have multiple resources that require a similar policy, you can group them together under one policy. This is possible by setting `validateResource` to an array of callback functions. The example below shows a case where we want to run similar checks against multiple types of resources.
+
+```typescript
+const elbLoggingEnabled: ResourceValidationPolicy = {
+    name: "elb-logging-enabled",
+    description: "Checks whether the Application Load Balancers and the Classic Load Balancers have logging enabled.",
+    enforcementLevel: "mandatory",
+    validateResource: [
+        validateTypedResource(aws.elasticloadbalancing.LoadBalancer, (loadBalancer, args, reportViolation) => {
+            if (loadBalancer.accessLogs === undefined || !loadBalancer.accessLogs.enabled) {
+                reportViolation("Elastic Load Balancer must have access logs enabled.");
+            }
+        }),
+        validateTypedResource(aws.elasticloadbalancingv2.LoadBalancer, (loadBalancer, args, reportViolation) => {
+            if (loadBalancer.accessLogs === undefined || !loadBalancer.accessLogs.enabled) {
+                reportViolation("Elastic Load Balancer must have access logs enabled.");
+            }
+        }),
+        validateTypedResource(aws.applicationloadbalancing.LoadBalancer, (loadBalancer, args, reportViolation) => {
+            if (loadBalancer.accessLogs === undefined || !loadBalancer.accessLogs.enabled) {
+                reportViolation("Application Load Balancer must have access logs enabled.");
+            }
+        }),
+    ],
+};
+```
+
+## Stack Validation Policies
+
+Policies of `StackValidationPolicy` are run against the stack as a whole. These policies are run _after_ all stack resources are registered and thus do not block an out-of-compliance resource from being created or modified, but do fail the preview or update. To avoid creating out-of-compliance resources, we recommend running a preview command before an update. However, there are cases where a policy requires a resource to exist in order to perform its validation, such as inspecting the resource's output properties that are only available to already provisioned resources (e.g. checking if a resource's `id` is referenced by another resource). In such cases, the policy can skip reporting violations during previews and only report violations during updates when the information it needs is available. Thus, the preview will succeed and the resources will be created or modified during the update, but the overall update operation will fail if any resources are in violation.
```

Comment:
```suggestion
Policies of `StackValidationPolicy` run against the whole stack. These policies are run _after_ all stack resources are registered and thus do not block an out-of-compliance resource from being created or modified, but do fail the preview or update. To avoid creating out-of-compliance resources, we recommend running a preview command before an update. 

However, there are cases where a resource must exist for a policy to validate it, such as inspecting the resource's output properties that are only available to already provisioned resources (e.g., checking if a resource's `id` is referenced by another resource). In such cases, the policy can skip reporting violations during previews and only report violations during updates when the information it needs is available. Thus, the preview can succeed. and the resources are created or modified during the update. The overall update operation fails if any resources are in violation.
```